sklearn/partial_dependence.py

```diff
@@ -0,0 +1,574 @@
+                                 est.__class__.__name__)
+        elif isinstance(est, ClassifierMixin):
+            try:
+                pdp_row = est.predict_proba(X_eval)
+            except:
+                raise ValueError('Call %s.fit before partial_dependence' %
+                                 est.__class__.__name__)
+            pdp_row = np.log(np.clip(pdp_row, 1e-16, 1))
+            pdp_row = np.subtract(pdp_row,
+                                  np.mean(pdp_row, 1)[:, np.newaxis])
+            pdp.append(np.mean(pdp_row, 0))
+        else:
+            raise ValueError('est must be a fitted regressor or classifier '
+                             'model.')
+    pdp = np.array(pdp).transpose()
+    if pdp.shape[0] == 2:
+        # Binary classification
+        pdp = pdp[1, :][np.newaxis]
+    elif len(pdp.shape) == 1:
+        # Regression
+        pdp = pdp[np.newaxis]
+    return pdp
+
+
+def _estimated_partial_dependence(est, target_variables, grid, X):
+    """Calculate the partial dependence of ``target_variables``.
+
+    The function will be calculated by calling the ``predict_proba`` method of
+    ``est`` for classification or ``predict`` for regression on the mean of
+    ``X``.
+
+    Parameters
+    ----------
+    est : BaseEstimator
+        A fitted classification or regression model.
+    target_variables : array-like, dtype=int
+        The target features for which the partial dependency should be
+        computed (size should be smaller than 3 for visual renderings).
+    grid : array-like, shape=(n_points, len(target_variables))
+        The grid of ``target_variables`` values for which the
+        partial dependency should be evaluated (either ``grid`` or ``X``
+        must be specified).
+    X : array-like, shape=(n_samples, n_features)
+        The data on which ``est`` was trained.
+
+    Returns
+    -------
+    pdp : array, shape=(n_classes, n_points)
+        The partial dependence function evaluated on the ``grid``.
+        For regression and binary classification ``n_classes==1``.
+    """
+    n_samples = grid.shape[0]
+    X_eval = np.tile(X.mean(0), [n_samples, 1])
+    for i, variable in enumerate(target_variables):
+        X_eval[:, variable] = grid[:, i]
+    if isinstance(est, RegressorMixin):
+        try:
+            pdp = est.predict(X_eval)
+        except:
+            raise ValueError('Call %s.fit before partial_dependence' %
+                             est.__class__.__name__)
+        pdp = pdp[np.newaxis]
+    elif isinstance(est, ClassifierMixin):
+        try:
+            pdp = est.predict_proba(X_eval)
+        except:
+            raise ValueError('Call %s.fit before partial_dependence' %
+                             est.__class__.__name__)
+        pdp = np.log(np.clip(pdp, 1e-16, 1))
+        pdp = np.subtract(pdp, np.mean(pdp, 1)[:, np.newaxis])
+        pdp = pdp.transpose()
+    else:
+        raise ValueError('est must be a fitted regressor or classifier model.')
```

Comment:
I'd have to look into it, at first thought it may get tricky to get a transformed X on which to construct the pdplot grid and calculate the function. Can you give an example of such a pipeline you would want to use, and where you'd want the pdplot to be calculated?